zucchini-ui-frontend/src/stepDefinitions/components/StepDefinitionsTable.jsx

```diff
@@ -0,0 +1,255 @@
+    if (event) {
+      event.preventDefault();
+    }
+    this.props.onClose();
+  };
+
+  render() {
+    const { show, occurrences, location } = this.props;
+    return (
+      <Modal bsSize='large' dialogClassName='modal-dialog' show={show} onHide={this.onCloseClick}>
+        <Modal.Header closeButton>
+          <Modal.Title>{'Variantes Connues'} <span className='step-definition-location'>{location}</span></Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <pre>{formatStepDefinitionVariants(occurrences)}</pre>
+        </Modal.Body>
+      </Modal>
+    );
+  }
+}
+
+function groupDistinctArgumentsByOffset(occurrences) {
+  const groupedArgs = {};
+  const knownOffsets = occurrences[0].info.arguments.map(arg => {
+    return arg.offset
+  });
+
+  if (knownOffsets.length > 0) {
+    occurrences.forEach(occurrence => {
+      occurrence.info.arguments.map(argument => {
+        const key = getClosestOffset(knownOffsets, argument.offset);
+        const value = argument.value;
+        if (groupedArgs[key] === undefined) {
+          groupedArgs[key] = [value];
+        } else {
+          // No value duplicates
+          if (!includes(groupedArgs[key], value)) {
+            groupedArgs[key].push(value);
+          }
+        }
+      })
+    });
+  }
+  return groupedArgs;
+}
+
+function getClosestOffset(knownOffsets, currentOffset) {
+  // Sometimes the offset returned by cucumber is slightly shifted due to non-capturing groups in regexes
+  // Ex: " this is a glue with non capturing groups?" -> The "s" in "groups" will shift ethe offset by one
+  if (includes(knownOffsets[currentOffset])) {
+    return currentOffset;
+  }
+  return knownOffsets.reduce((prev, curr) => Math.abs(curr - currentOffset) < Math.abs(prev - currentOffset) ? curr : prev);
+}
+
+function formatVariantsBlock(variants, variantStyle) {
+  return variants.map(variant => {
+    return <StepDefinitionsTableHighlightedTerm key={variant} text={variant} styleClass={variantStyle} />
+  });
+}
+
+function formatStepDefinition(stepDefinition) {
+
+  const keyword = stepDefinition.info.keyword;
+  const definition = truncate(stepDefinition.info.name, { 'length': 300 });
+  const args = stepDefinition.info.arguments;
+
+  const formattedText = [];
+  formattedText.push(<span key={Math.random()}><StepDefinitionsTableHighlightedTerm text={keyword} styleClass='step-definition-keyword' />&nbsp;</span>);
+
+  let startIdx = 0;
+  if (args !== undefined) {
+    args.forEach((arg, index) => {
+      formattedText.push(definition.substr(startIdx, arg.offset - startIdx));
+      formattedText.push(<StepDefinitionsTableHighlightedTerm key={index} text={arg.value} styleClass='step-definition-argument' />);
+      startIdx = arg.offset + arg.value.length;
+    });
+    formattedText.push(definition.substr(startIdx));
+  } else {
+    formattedText.push(definition)
+  }
+  return formattedText;
+}
+
+function formatStepDefinitionVariants(occurrences) {
```

Comment:
You should generate HTML directly and not use React. Look at the ElementInfo component for that.